### PR TITLE
Make precision matrix computation in mvn stable

### PR DESCRIPTION
**Pull request by fehiepsi**
As explained in [here](https://github.com/uber/pyro/issues/953#issuecomment-377493414), the current computation for precision matrix which uses the inverse of covariance matrix might lead to some instability problem. I think that it is better to compute prec matrix based on scale_tril.

### CODE REVIEW

**Comment by apaszke** [comment]
cc: @fritzo

File: torch/distributions/multivariate_normal.py [hunk 1]
```diff
@@ -159,8 +159,9 @@ def covariance_matrix(self):
     @lazy_property
     def precision_matrix(self):
         # TODO: use `torch.potri` on `scale_tril` once a backwards pass is implemented.
-        flat_conv = self.covariance_matrix.reshape((-1,) + self._event_shape * 2)
-        flat_precision = torch.stack([C.inverse() for C in flat_conv], 0)
+        scale_tril_inv = _batch_inverse(self.scale_tril)
+        flat_scale_tril_inv = self.scale_tril.reshape((-1,) + self._event_shape * 2)
+        flat_precision = torch.stack([C.t().matmul(C) for C in flat_scale_tril_inv], 0)
```
Comment (SsnL) [inline torch/distributions/multivariate_normal.py]: I think we can use `.bmm` here

**Review by fritzo — APPROVED** [review]
LGTM.

I guess an unintended consequence of #5998 is that now `validate_args=True` triggers computation of the `precision_matrix` even if only `scale_tril` or `covariance_matrix` is specified. Maybe we should try to avoid this extra computation.

File: torch/distributions/multivariate_normal.py [hunk 2]
```diff
@@ -159,8 +159,10 @@ def covariance_matrix(self):
     @lazy_property
     def precision_matrix(self):
         # TODO: use `torch.potri` on `scale_tril` once a backwards pass is implemented.
-        flat_conv = self.covariance_matrix.reshape((-1,) + self._event_shape * 2)
-        flat_precision = torch.stack([C.inverse() for C in flat_conv], 0)
+        scale_tril_inv = _batch_inverse(self.scale_tril)
+        flat_scale_tril_inv = self.scale_tril.reshape((-1,) + self._event_shape * 2)
```
Comment (vishwakftw) [inline torch/distributions/multivariate_normal.py]: I think this is the error. I believe it is supposed to be `self.scale_tril_inv`.

Comment (vishwakftw) [reply torch/distributions/multivariate_normal.py]: Fixed in #6172 .

**Pull request merged**